tokio-fs/tests/file.rs

```diff
@@ -41,7 +41,9 @@ fn read_write() {
         let contents = contents.clone();
 
         File::create(file_path)
-            .and_then(move |file| io::write_all(file, contents))
+            .and_then(|file| file.metadata())
+            .inspect(|(metadata, _)| assert!(metadata.is_file()))
+            .and_then(move |(_, file)| io::write_all(file, contents))
```

Comment:
@carllerche It feels a bit "unbalanced" to have `metadata` return a `(Metadata, File)` but `io::write_all` return a `(File, Whatever)`. Should I flip it to return a `(File, Metadata)`?

Reply:
> Should I flip it to return a `(File, Metadata)`

I did this. It just feels more correct.